src/pages/Character.tsx

```diff
@@ -116,7 +116,16 @@ const Character = () => {
 
   const handleInputChange = (e) => {
     const { name, value } = e.target;
-    setCharacterData((prevData) => ({ ...prevData, [name]: value }));
+    const newValue =
+      name.includes('level') ||
+      name.includes('health') ||
+      stats.some((stat) => stat.name === name)
+        ? isNaN(value)
+          ? 0
+          : Number(value)
+        : value;
```

Comment:
To bym przeniósł do osobnej funkcji, nazwał ją może processValue lub coś tego typu.

Funkcja będzie wyglądać mniej więcej tak (dziwnie sformatowane to jest tutaj):


  const processValue = (name, value) => {
    const includesKeywords = name.includes("level") || name.includes("health");
    const containsStatName =  stats.some((stat) => stat.name === name);

    if (includesKeywords  || containsStatName) {
      return isNaN(value) ? 0 : +value;
    }

    return value;
  };